test/backend/lazy_load_test.rb

```diff
@@ -0,0 +1,107 @@
+require 'test_helper'
+
+class I18nBackendLazyLoadTest < I18n::TestCase
+  def setup
+    super
+    @backend  = I18n.backend = I18n::Backend::LazyLoad.new
+    I18n.load_path = [locales_dir + '/en.yml', locales_dir + '/fr.yml']
```

Comment:
To avoid double `/` if `locales_dir` have a trailing slash:
```suggestion
    I18n.load_path = [File.join(locales_dir, 'en.yml'), File.join(locales_dir, 'fr.yml')]
```